shipyard/src/main/java/org/jclouds/shipyard/features/ServiceKeysApi.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.shipyard.features;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.binders.BindToJsonPayload;
+import org.jclouds.shipyard.domain.servicekeys.CreateServiceKey;
+import org.jclouds.shipyard.domain.servicekeys.DeleteServiceKey;
+import org.jclouds.shipyard.domain.servicekeys.ServiceKey;
+import org.jclouds.shipyard.filters.ServiceKeyAuthentication;
+
+@Consumes(MediaType.APPLICATION_JSON)
+@RequestFilters({ ServiceKeyAuthentication.class })
+@Path("/api/servicekeys")
+public interface ServiceKeysApi {
+
+   @Named("servicekeys:list")
+   @GET
+   List<ServiceKey> listServiceKeys();
+   
+   @Named("servicekeys:create")
+   @POST
+   ServiceKey createServiceKey(@BinderParam(BindToJsonPayload.class) CreateServiceKey createServiceKey);
+   
+   @Named("servicekeys:delete")
+   @DELETE
+   void deleteServiceKey(@BinderParam(BindToJsonPayload.class) DeleteServiceKey deleteServiceKey);
```

Comment:
Looking at this with more detail, it makes not sense to have an object just to hold a single value. Change these methods to receive a String, and configure the annotations to automatically populate it to the payload (look at [any Profitbricks API](https://github.com/jclouds/jclouds-labs/blob/master/profitbricks/src/main/java/org/jclouds/profitbricks/features/DataCenterApi.java#L64-L69) for an example) and remove the Create and Delete service key classes.